### PR TITLE
[MLIR][Arith] Fold on extension of FP constants using arith.extf

### PR DESCRIPTION
**Pull request by victor-eds**
It is safe to fold when extending, as we will not lose precision.

Signed-off-by: Victor Perez <victor.perez@codeplay.com>

### CODE REVIEW

File: mlir/lib/Dialect/Arith/IR/ArithOps.cpp [hunk 1]
```diff
@@ -1100,6 +1100,18 @@ LogicalResult arith::ExtSIOp::verify() {
...
+OpFoldResult arith::ExtFOp::fold(ArrayRef<Attribute> operands) {
+  assert(operands.size() == 1 && "unary operation takes one operand");
+
+  auto constOperand = operands.front().dyn_cast_or_null<FloatAttr>();
+  if (!constOperand)
+    return {};
+
+  // Convert to target type via 'double'.
+  return FloatAttr::get(getType(), constOperand.getValue().convertToDouble());
+}
```
Comment (victor-eds) [inline mlir/lib/Dialect/Arith/IR/ArithOps.cpp:1113]: This differs from `arith::TruncFOp::fold`, as there they check whether precision is lost.

**Review by whitneywhtsang — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Comment by whitneywhtsang** [comment]
@victor-eds Can we upstream this change to avoid merge failures?

**Comment by victor-eds** [comment]
> @victor-eds Can we upstream this change to avoid merge failures?

https://reviews.llvm.org/D144251

We will still get, as the problem is the input type has changed to `FoldAdaptor`

**Comment by whitneywhtsang** [comment]
> > @victor-eds Can we upstream this change to avoid merge failures?
> 
> https://reviews.llvm.org/D144251
> 
> We will still get, as the problem is the input type has changed to `FoldAdaptor`

Right, I only noticed this as I am working on a merge fix which this is one of the changes needed. 😄

**Comment by victor-eds** [comment]
> > > @victor-eds Can we upstream this change to avoid merge failures?
> > 
> > 
> > https://reviews.llvm.org/D144251
> > We will still get, as the problem is the input type has changed to `FoldAdaptor`
> 
> Thanks for putting up an upstream review! Right, I only noticed this as I am working on a merge fix which this is one of the changes needed. smile

I've just got it [merged](https://github.com/llvm/llvm-project/commit/6fa26f60c55d4d7ba0bfc5353a8f7583da9d4b9a)

**Pull request merged**